core/blockchain.go

```diff
@@ -1993,14 +1993,10 @@ func (bc *BlockChain) recoverAncestors(block *types.Block) (common.Hash, error)
 }
 
 // collectLogs collects the logs that were generated or removed during
-// the processing of the block that corresponds with the given hash.
-// These logs are later announced as deleted or reborn.
-func (bc *BlockChain) collectLogs(hash common.Hash, removed bool) []*types.Log {
-	number := bc.hc.GetBlockNumber(hash)
-	if number == nil {
-		return nil
-	}
-	receipts := rawdb.ReadReceipts(bc.db, hash, *number, bc.chainConfig)
+// the processing of a block. These logs are later announced as deleted or reborn.
+func (bc *BlockChain) collectLogs(b *types.Block, removed bool) []*types.Log {
+	receipts := rawdb.ReadRawReceipts(bc.db, b.Hash(), b.NumberU64())
+	receipts.DeriveFields(bc.chainConfig, b.Hash(), b.NumberU64(), b.Transactions())
```

Comment:
This is a behavioural change. How come you are adding this here?

Reply:
No it's not a behavioral change! The previous version used `ReadReceipts`, which calls `DeriveFields` internally after loading the block from disk. My change is to avoid loading the block since we already have it in memory here.